ouroboros-consensus/docs/ReleaseProcess.md

```diff
@@ -0,0 +1,140 @@
+# How Consensus Makes Releases
+
+This document explains how the Consensus team uses Pull Requests (PRs), version numbers, and branches to prepare releases of our packages.
+
+Let us assume our repository contains two packages FOO and BAR.
+Two is enough to explain the complexities.
+The resulting rules can be generalized to any number of packages, as explained at the end of the document.
+
+*Remark*.
+If you disagree with the decisions in this document, please consider reviewing our brainstorming notes and discussions, available at [this document's PR](https://github.com/input-output-hk/ouroboros-network/pull/4207).
+Even if you choose not to do so, please share your concern with us.
+
+## Notation
+
+In this document, we will assume each version number is a MAJOR.MINOR.PATCH triple.
+This is easy to generalize to other conventional notations, such as [the PVP](https://pvp.haskell.org/)'s MAJOR.MAJOR.MINOR.PATCH quadruples.
+
+We refer throughout this document to _the main branch_ (aka `master`, aka `main`).
+
+If you search for "RULE:" in this document, you'll find the key statements.
+
+## Rules for Branches
+
+The Consensus team will sometimes be maintaining multiple MAJOR.MINOR versions of the same package simultaneously.
+EG We might release FOO-2.4.1 on Monday, release FOO-2.3.7 on Wednesday, and release FOO-2.4.2 on Friday, etc.
+
+- Whenever we're maintaining only the one greatest-ever MAJOR.MINOR version of a package, then all work on that package will happen on the main branch.
```

Comment:
Something like this? 

```text
            FOO-2.3.6 --- FOO-2.3.7 <- release branch 
             /                                            
-------FOO-2.3.? ------  .... ------ FOO-2.4.1  
                                        ^
                                        |
                                   main branch
```

Assuming `FOO-2.4.1` is the greatest-ever version, if we have to maintain this version, then we develop on `main`. If we have to maintain `2.3.7`, then we work on the release branch.

Reply:
> If we have to maintain 2.3.7, then we work on the release branch.

If necessary, yes (see my reply to your other comment below). But hopefully that "work on the release branch" will just be cherry-picking some commits from the main branch.

Reply:
I pushed up a longer example. Does it suffice for this?

Reply:
I guess you refer to [this](https://github.com/input-output-hk/ouroboros-network/pull/4207/commits/02482cf7fbabf1b6e4441dd6d730f36c121bb45e#diff-a3b11ca853f39e9e181a2ad2868b36c39f946100f19907a46e41607456b48cd9R148) right? That example is really helpful, I just want to confirm my assumption: at the end of the example we'll end up with this commit structure, right?

```
F   J
|   |
E   I 
|   | 
D   H  
|   | 
C   G
| /
B
|
A
```